stockflux-watchlist/src/components/Watchlist/Watchlist.js

```diff
@@ -10,13 +10,34 @@ function Watchlist() {
   const [dragOverIndex, setDragOverIndex] = useState(null);
   const [dragStartClientY, setDragStartClientY] = useState(null);
   const [cardHeight, setCardHeight] = useState(null);
-  const [watchlist, setWatchlist] = useState([
-    'AAPL',
-    'AAP',
-    'CC',
-    'MS',
-    'JPS'
-  ]);
+  const [watchlist, setWatchlist] = useState([]);
+  const [isInitialised, setIsInitialised] = useState(false);
+
+  useEffect(() => {
+    setIsInitialised(false);
+    const initialWatchlist = ['AAPL', 'AAP', 'CC', 'MS', 'JPS'];
+    const localStorageWatchlist = JSON.parse(localStorage.getItem('watchlist'));
+    const tempWatchlist =
+      !localStorageWatchlist || localStorageWatchlist.length === 0
+        ? initialWatchlist
+        : localStorageWatchlist;
+    persistWatchlist(tempWatchlist);
+    setIsInitialised(true);
+  }, []);
+
+  useEffect(() => {
```

Comment:
This will have a race condition if the app is opened via a raise intent

Reply:
Correct, fixed -> 4f071f107e9fccdded865ae548a372ef7bc64b80